### PR TITLE
Fixing #580 (Fully qualify common types in error messages)

### PR DESCRIPTION
**Pull request by aaronjeline**
## Description of changes
The error message for `UndeclaredCommonTypes` now reports fully qualified type names, which is less confusing
## Issue #, if available
#580 
## Checklist for requesting a review

The change in this PR is (choose one, and delete the other options):

- [X] A bug fix or other functionality change requiring a patch to `cedar-policy`.

I confirm that this PR (choose one, and delete the other options):

- [x] Updates the "Unreleased" section of the CHANGELOG with a description of my change (required for major/minor version bumps).
- [ ] Does not update the CHANGELOG because my change does not significantly impact released code.

I confirm that [`cedar-spec`](https://github.com/cedar-policy/cedar-spec) (choose one, and delete the other options):

- [X] Does not require updates because my change does not impact the Cedar formal model or DRT infrastructure.

## Disclaimer

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cedar-policy-validator/src/err.rs [hunk 1]
```diff
@@ -192,3 +192,93 @@ pub enum UnsupportedFeature {
...
+        );
+        let schema = ValidatorSchema::from_json_value(src, Extensions::all_available());
+        assert!(schema.is_err(), "This schema should not parse");
+        match schema.unwrap_err() {
+            SchemaError::UndeclaredCommonTypes(types) => {
+                assert_eq!(types.len(), 1);
+                assert!(types.contains("Demo::id"));
+            }
+            other => panic!("Wrong error: {:?}", other),
+        }
```
Comment (john-h-kastner-aws) [inline cedar-policy-validator/src/err.rs:242]:
```rust
assert_matches!(schema, Err(SchemaError::UndeclaredCommonTypes(types)) =>
    assert_eq!(types, HashSet::from(["Demo::id".to_string()])));
```

File: cedar-policy-validator/src/err.rs [hunk 2]
```diff
@@ -192,3 +192,93 @@ pub enum UnsupportedFeature {
...
+                      "type": "String"
+                    },
+                  },
+                  "entityTypes": {},
+                  "actions": {}
+                }
+              }
+        );
+        let schema = ValidatorSchema::from_json_value(src, Extensions::all_available());
+        assert!(schema.is_err(), "This schema should not parse");
```
Comment (john-h-kastner-aws) [inline cedar-policy-validator/src/err.rs:275]:
```rust
assert_matches!(schema, Err(SchemaError::UndeclaredCommonTypes(types)) =>
    assert_eq!(types, HashSet::from(["Demo::id".to_string()])));
```

File: cedar-policy-validator/src/err.rs [hunk 3]
```diff
@@ -192,3 +192,93 @@ pub enum UnsupportedFeature {
     #[error("action declared with attributes: [{}]", .0.iter().join(", "))]
     ActionAttributes(Vec<String>),
 }
+
+// PANIC SAFETY testing code
+#[allow(clippy::panic)]
+#[cfg(test)]
+mod test {
```
Comment (john-h-kastner-aws) [inline cedar-policy-validator/src/err.rs:199]: nit: most schema parsing tests live in `schema.rs`

**Review by john-h-kastner-aws — COMMENTED** [review]
~~Does this also require a change to the SchemaError enum in `cedar-policy`?~~
Edit: it doesn't

File: cedar-policy-validator/src/err.rs (hunk 3 above)
Comment (aaronjeline) [reply cedar-policy-validator/src/err.rs:199]: Ah sure thing, I'll move them

File: cedar-policy-validator/src/err.rs (hunk 1 above)
Comment (aaronjeline) [reply cedar-policy-validator/src/err.rs:242]: Ooh `assert_matches!` was more powerful than I realized

**Review by john-h-kastner-aws — APPROVED** [review]

**Review by shaobo-he-aws — APPROVED** [review]

**Pull request merged**